proxy/src/http/sql_over_http.rs

```diff
@@ -189,27 +190,41 @@ pub async fn handle(
     conn_pool: Arc<GlobalConnPool>,
     session_id: uuid::Uuid,
 ) -> Result<Response<Body>, ApiError> {
-    let result = handle_inner(request, sni_hostname, conn_pool, session_id).await;
+    let result = tokio::time::timeout(
+        tokio::time::Duration::from_secs(HTTP_CONNECTION_TIMEOUT_SECS),
+        handle_inner(request, sni_hostname, conn_pool, session_id),
+    )
+    .await;
 
     let mut response = match result {
-        Ok(r) => r,
-        Err(e) => {
-            let message = format!("{:?}", e);
-            let code = match e.downcast_ref::<tokio_postgres::Error>() {
-                Some(e) => match e.code() {
-                    Some(e) => serde_json::to_value(e.code()).unwrap(),
+        Ok(r) => match r {
+            Ok(r) => r,
+            Err(e) => {
+                let message = format!("{:?}", e);
+                let code = match e.downcast_ref::<tokio_postgres::Error>() {
+                    Some(e) => match e.code() {
```

Comment:
I know that you are only moving this code around but can you maybe adopt `and_then` here? That would cut down on the right ward drift a bit.

Reply:
Done.